bundles/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/internal/handler/EvohomeHeatingZoneHandler.java

```diff
@@ -82,16 +89,115 @@ public void handleCommand(ChannelUID channelUID, Command command) {
                 if (EvohomeBindingConstants.ZONE_SET_POINT_CHANNEL.equals(channelId)
                         && command instanceof DecimalType) {
                     double newTemp = ((DecimalType) command).doubleValue();
-                    if (newTemp == CANCEL_SET_POINT_OVERRIDE) {
+                    // Get a local copy of schedule when needed, saves on api calls
+                    DailySchedules schedule = getDailySchedules();
+                    double expectedTemp = getCurrentSetPoint(schedule);
+                    if (newTemp == CANCEL_SET_POINT_OVERRIDE || newTemp == expectedTemp) {
                         bridge.cancelSetPointOverride(getEvohomeThingConfig().id);
+                        updateState(EvohomeBindingConstants.ZONE_SET_POINT_CHANNEL, new DecimalType(expectedTemp));
+                        return;
                     } else if (newTemp < 5) {
                         newTemp = 5;
                     }
                     if (newTemp >= 5 && newTemp <= 35) {
-                        bridge.setPermanentSetPoint(getEvohomeThingConfig().id, newTemp);
+                        int overrideMode = getOverrideMode();
+                        switch (overrideMode) {
+                            case EvohomeBindingConstants.SETPOINT_OVERRIDE_TEMPORARY_SCHEDULE_NEXT:
+                                bridge.setTemporarySetPoint(getEvohomeThingConfig().id, newTemp,
+                                        getNextSetPointChange(schedule));
+                                break;
+                            case EvohomeBindingConstants.SETPOINT_OVERRIDE_TEMPORARY_TIME_ADD:
+                                bridge.setTemporarySetPoint(getEvohomeThingConfig().id, newTemp,
+                                        LocalDateTime.now().plusMinutes(getOverrideTime()));
+                                break;
+                            default:
+                                bridge.setPermanentSetPoint(getEvohomeThingConfig().id, newTemp);
+                                break;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private DailySchedules getDailySchedules() {
+        return this.getEvohomeBridge().getZoneSchedule("temperatureZone", this.getId());
+    }
```

Comment:
"`Schedules`" plural, though it seems to be the schedule of one day.
Also, magic string, which seems to be used only here. Please move inside `getZoneSchedule` as a constant.

Reply:
this is the schedule object which holds the schedule for the entire week, so multiple "daily schedule"s so I think i'd argue for the plural?